### PR TITLE
CLDR-13853 unit test for VettingViewer English Changed

### PR DESCRIPTION
**Pull request by srl295**
- use JUnit5 for a modern test experience
- add a VSCode settings file for test execution

CLDR-13853

depends on #1343

### CODE REVIEW

File: tools/cldr-code/src/test/java/org/unicode/cldr/util/TestVettingViewer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * Also see {@link org.unicode.cldr.unittest.TestUtilities}
+ */
+class TestVettingViewer {
+    static final String CENTURY_NARROW_DISPLAY_NAME = "//ldml/units/unitLength[@type=\"narrow\"]/unit[@type=\"duration-century\"]/displayName";
+    @Test
+    void testOutdatedPaths() {
+        String previous = VettingViewer.getOutdatedPaths().getPreviousEnglish(CENTURY_NARROW_DISPLAY_NAME);
+        assertNotEquals(OutdatedPaths.NO_VALUE, previous, "expected prev english to be present for " + CENTURY_NARROW_DISPLAY_NAME);
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/util/TestVettingViewer.java:27]: @macchiati perhaps this is to be expected, though?

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/util/TestVettingViewer.java:27]: It turns out that the English value was not explicitly set until v37 (aliased value), so it is predated in some locales.

**Review by macchiati — APPROVED** [review]
Looks great, and we can build on the scaffolding in the future.

File: tools/cldr-code/src/test/java/org/unicode/cldr/util/TestVettingViewer.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-code/src/test/java/org/unicode/cldr/util/TestVettingViewer.java:27]: this particular test fails (for the reason you gave)  I can omit it, but keep the other test function.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "java.test.config": {
+        "env": {
+            "CLDR_DIR": "${workspaceRoot}",
+            "CLDR_ENVIRONMENT": "UNITTEST"
+        }
+    }
+}
```
Comment (btangmu) [inline .vscode/settings.json:8]: is this the first file added to the repo that's specific to VS Code? how are you using it?

I've been using VS Code for editing js files, but not for java or actually running any code

**Review by btangmu — COMMENTED** [review]
`Error:    TestVettingViewer.testOutdatedPaths:27 expected prev english to be present for //ldml/units/unitLength[@type="narrow"]/unit[@type="duration-century"]/displayName ==> expected: not equal but was: <�>`
-- is that what you were referring to ("this particular test fails")?

**Comment by srl295** [comment]
> `Error:    TestVettingViewer.testOutdatedPaths:27 expected prev english to be present for //ldml/units/unitLength[@type="narrow"]/unit[@type="duration-century"]/displayName ==> expected: not equal but was: <�>`
> 
> -- is that what you were referring to ("this particular test fails")?

Yes. It's expected , so I'll remove that test case.

File: .vscode/settings.json (hunk 2 above)
Comment (srl295) [reply .vscode/settings.json:8]: it enables "run this test" or "debug this test" as a button right next to any test function.

**Comment by macchiati** [comment]
Where does this 'button' show up?

Mark

**Comment by srl295** [comment]
> Where does this 'button' show up? Mark
> […](#)

in visual studio code, next to the test class definition (for running the entire class)  and also the test function.

image

Eclipse should have similar,  "Run As -> JUnit Test Case"

**Comment by srl295** [comment]
i removed the failing test case for the U+FFFD, so the test should pass now. 

The test will show as "skipped" if it didn't find any english changed to test, this won't fail but will show up in the overall logs.  Otherwise it will make sure that everything "English Changed" is present with a non-U+FFFD prior value.

**Comment by macchiati** [comment]
Sounds good, thanks.
Mark

**Review by btangmu — APPROVED** [review]

**Pull request merged**